sg_otio/sg_cut_track_writer.py

```diff
@@ -907,18 +907,10 @@ def _get_shot_payload(self, clip_group, sg_project, sg_linked_entity, sg_user=No
         if True:
             shot_payload[sfg.has_effects] = clip_group.has_effects
             shot_payload[sfg.has_retime] = clip_group.has_retime
-        if sfg.absolute_cut_order:
-            # TODO: maybe create Shot fields config before so that
-            #       we can get the field without an extra query?
-            sg_entity = self._sg.find_one(
-                sg_linked_entity["type"],
-                [["id", "is", sg_linked_entity["id"]]],
-                [sfg.absolute_cut_order]
-            )
-            entity_cut_order = sg_entity.get(sfg.absolute_cut_order)
-            if entity_cut_order:
-                absolute_cut_order = 1000 * entity_cut_order + clip_group.index
-                shot_payload[sfg.absolute_cut_order] = absolute_cut_order
+        if sfg.absolute_cut_order and sg_linked_entity.get(sfg.absolute_cut_order):
```

Comment:
Should we raise an error if it's not there? It should be shouldn't it? Should this covered by unit tests so we can guarantee that it is always there if needed?

Reply:
You mean the field or the field value? Before, we use to add it to the payload only if the field existed and there was a value. I can change the behavior but I have to first understand if you mean the field or the value.

- If you mean the field, then we could fail if it does not exist, but I believe some people don't use absolute cut order, so they don't need it to exist, and the tool would still work unless you choose the absolute tc2frame mapping... Maybe that's what you mean, raise an error in this case?
- If you mean the value, TBH I'm not sure if the linked entity must always have an absolute cut order defined. I'd have to dig more.

Reply:
The field. If I understand well the `if sig.absolute_cut_order` test indicates that the user uses absolute cut order? So the field should be there? For the value, the behaviour should be the same as before.

Reply:
nope, we have two kinds of settings:

- `SGSettings`, which have a `timecode_in_to_frame_mapping_mode` that indicates if it's absolute/relative/automatic
- `SGShotFieldsConfig` which is just which Shot fields to use for the payload (smart cut fields or not basically), and this is the code of the `absolute_cut_order` method:

```
@property
    def absolute_cut_order(self):
        """
        Get the field for absolute cut order, if any.

        :returns: A str or ``None``.
        """
        if _ABSOLUTE_CUT_ORDER_FIELD in self.shot_schema:
            return _ABSOLUTE_CUT_ORDER_FIELD
        return None
```

Reply:
I checked the old code: 

- docstring: https://github.com/GPLgithub/tk-multi-importcut/blob/fd0a946de26990f24a5c5993a2d2b19a620872ba/python/tk_multi_importcut/edl_cut.py#L1351
- the test: https://github.com/GPLgithub/tk-multi-importcut/blob/fd0a946de26990f24a5c5993a2d2b19a620872ba/python/tk_multi_importcut/edl_cut.py#L1442
- not only shots, cut items too: https://github.com/GPLgithub/tk-multi-importcut/blob/fd0a946de26990f24a5c5993a2d2b19a620872ba/python/tk_multi_importcut/edl_cut.py#L1904

it does seem inconsistent the way we do it for cut items though, we should use the same approach everywhere, shouldn't we? https://github.com/GPLgithub/sg-otio/blob/bd4b011a06f3f83d85f774876e32ec081092666d/sg_otio/sg_cut_track_writer.py#L391
not sure it's worth doing it though, what do you think?

Reply:
Do we have anything which enforces the absolute cut order field in the schema if some setting are requiring it?
And looking at the property code, `sg_linked_entity.get(sfg.absolute_cut_order)` is not needed, right? If `sfg.absolute_cut_order` returns anything, it should be available in the `sg_linked_entity` (with a possible empty value).

Reply:
As far as I can see, there is no setting that uses that field, and we don't use it to calculate anything, we just fill it in if it's available in the schema. Which is what was already done before.
And we weren't adding it to the payload when `sg_linked_entity.get(sfg.absolute_cut_order)` was not available, which is why I guess it's like this here, I'm guessing this field is an int field and putting `None` as a value makes the code fail?

Reply:
Don't know, we're spending too much time on this. I initially approved the PR, so approving it again and let's move forward.